testing/integration_tests/test_graph.py

```diff
@@ -0,0 +1,69 @@
+import pytest, json
+from src.utils.graph import Graph
+import networkx as nx
+
+# Load sample response data
+# This file path might break with automated testing
+with open('testing/integration_tests/test_response.json') as f:
+    sample_data = json.load(f)
+
+# Create a truth graph to compare
+known_graph = nx.DiGraph()
+start = 'Boston%2CMA'
+end = 'Lexington%2CMA'
+se_node = 'Boston%2CMA&Lexington%2CMA'
+midpoints = ['Charleston%2CMA', 'Concord%2CMA', 'Sudbury%2CMA', 'Westwood%2CMA', 'Arlington%2CMA', 'Needham%2CMA']
+origins = [start] + midpoints
+dests = [end] + midpoints
+
+def test_graph():
+    # Create test graph
+    graph = Graph(start=start, end=end)
+
+    # Compare
+    assert graph._se_node == se_node
+
+def test_add_nodes():
+    # Create test graph
+    graph = Graph(start=start, end=end)
+    graph.add_nodes(midpoints=midpoints)
+
+    # Add nodes to truth graph
+    known_graph.add_node(se_node)
```

Comment:
These variables should probably be included in each individual test, which I know is annoying but I think rewriting all the variables is good practice in tests
Calls back to the whole "everything you need to know about the test should be in the test function"
At the very least, need to reset `known_graph` in case tests are run multiple times or something like that